src/components/UserConsent/index.tsx

```diff
@@ -107,14 +108,38 @@ const UserConsent: FunctionComponent<StepComponentBaseProps> = ({
     sdkContainer && unmountComponentAtNode(sdkContainer)
   }
 
+  const onContentLoadSuccess = (html: string) => {
+    setContentLoadError(false)
+    setConsentHtml(html)
+  }
+
+  const onContentLoadFailed = (err: string) => {
+    console.error(err)
+    setContentLoadError(true)
+  }
+
   useEffect(() => {
     new Promise<string>((resolve, reject) => {
       getConsentFile(resolve, reject)
     })
-      .then((html) => setConsentHtml(html))
-      .catch((err) => console.error(err))
+      .then(onContentLoadSuccess)
+      .catch(onContentLoadFailed)
   }, [])
 
+  if (isContentLoadError) {
+    return (
+      <ReloadContent
+        onPrimaryButtonClick={() => {
+          new Promise<string>((resolve, reject) => {
+            getConsentFile(resolve, reject)
+          })
+            .then(onContentLoadSuccess)
+            .catch(onContentLoadFailed)
```

Comment:
Fetching consent file code was duplicated here. We can move it out and use `async/await` for better readability

```suggestion
  const fetchConsentFile = () =>
    new Promise<string>((resolve, reject) => {
      getConsentFile(resolve, reject)
    })
      .then(onContentLoadSuccess)
      .catch(onContentLoadFailed)

  useEffect(() => {
    fetchConsentFile()
  }, [])

  if (isContentLoadError) {
    return <ReloadContent onPrimaryButtonClick={fetchConsentFile} />
  }
```

Reply:
Nice! Was trying to reduce the duplication overall but wasn't sure how to for this bit.